### PR TITLE
Remove Ruby 2.5

### PR DESCRIPTION
**Pull request by nicknovitski**
###### Motivation for this change

Ruby 2.5.x will reach its support end-of-life on March 30 of this year, so the next stable release of nixos shouldn't include it. 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- ~[ ] Ensured that relevant documentation is up to date~
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
This can go straight to master.

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -563,7 +563,6 @@ mapAliases ({
   ruby_2_3 = throw "ruby_2_3 was deprecated on 2019-09-06: use a newer version of ruby";
   ruby_2_4_3 = throw "ruby_2_4_3 was deprecated on 2018-02-13: use a newer version of ruby";
   ruby_2_4 = throw "ruby_2_4 was deprecated in 2019-12: use a newer version of ruby";
-  ruby_2_5_0 = throw "ruby_2_5_0 was deprecated on 2018-02-13: use a newer version of ruby";
```
Comment (SuperSandro2000) [inline pkgs/top-level/aliases.nix:566]: Please revert that change. Otherwise people run into eval errors.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -7399,8 +7399,7 @@ in
 
   redir = callPackage ../tools/networking/redir { };
 
-  # failed to build websocket-driver gem with ruby 2.6, so sticking to 2.5 for now
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:7402]: Is this fixed?

File: pkgs/top-level/aliases.nix (hunk 1 above)
Comment (nicknovitski) [reply pkgs/top-level/aliases.nix:566]: Oh, of course, I wasn't thinking.

Should I add similar deprecation error lines to this file for `ruby_2_5` and `rubyPackages_2-5`?

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (nicknovitski) [reply pkgs/top-level/all-packages.nix:7404]: Looks like they passed on ofborg.

**Review by Izorkin — APPROVED** [review]

File: pkgs/top-level/aliases.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/top-level/aliases.nix]: Yes please.

**Review by SuperSandro2000 — APPROVED** [review]

**Review by kalbasit — APPROVED** [review]

**Pull request merged**